lib/index.js

```diff
@@ -1,216 +1,234 @@
-var assert = require('assert')
-var clone = require('clone')
-var debug = require('debug')('analytics-node')
-var noop = function () {}
-var request = require('superagent')
-var uid = require('crypto-token')
-var version = require('../package.json').version
-var extend = require('lodash/extend')
-var validate = require('@segment/loosely-validate-event')
-var removeSlash = require('remove-trailing-slash')
-
-global.setImmediate = global.setImmediate || process.nextTick.bind(process)
+'use strict'
 
-/**
- * Expose an `Analytics` client.
- */
+const assert = require('assert')
+const removeSlash = require('remove-trailing-slash')
+const validate = require('@segment/loosely-validate-event')
+const request = require('superagent')
+const debug = require('debug')('analytics-node')
+const uid = require('crypto-token')
+const version = require('../package').version
 
-module.exports = Analytics
+const setImmediate = global.setImmediate || process.nextTick.bind(process)
+const noop = () => {}
 
 /**
- * Initialize a new `Analytics` with your Segment project's `writeKey` and an
- * optional dictionary of `options`.
+ * Get an error from a `res`.
  *
- * @param {String} writeKey
- * @param {Object} [options] (optional)
- *   @property {Number} flushAt (default: 20)
- *   @property {Number} flushAfter (default: 10000)
- *   @property {String} host (default: 'https://api.segment.io')
+ * @param {Object} res
+ * @return {String}
  */
 
-function Analytics (writeKey, options) {
-  if (!(this instanceof Analytics)) return new Analytics(writeKey, options)
-  assert(writeKey, 'You must pass your Segment project\'s write key.')
-  options = options || {}
-  this.queue = []
-  this.writeKey = writeKey
-  this.host = removeSlash(options.host || 'https://api.segment.io')
-  this.flushAt = Math.max(options.flushAt, 1) || 20
-  this.flushAfter = options.flushAfter || 10000
-}
+const error = res => {
+  if (!res.error) {
+    return
+  }
 
-/**
- * Send an identify `message`.
- *
- * @param {Object} message
- * @param {Function} [fn] (optional)
- * @return {Analytics}
- */
+  const body = res.body
+  const msg = (body.error && body.error.message) || `${res.status} ${res.text}`
 
-Analytics.prototype.identify = function (message, fn) {
-  validate(message, 'identify')
-  this.enqueue('identify', message, fn)
-  return this
+  return new Error(msg)
 }
 
-/**
- * Send a group `message`.
- *
- * @param {Object} message
- * @param {Function} [fn] (optional)
- * @return {Analytics}
- */
+class Analytics {
```

Comment:
The next release would be a major anyway, since we no longer include browserify build and also because of https://github.com/segmentio/analytics-node/pull/105, which can also break someone's code.